pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -242,3 +248,31 @@ func panicHandler(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+func setDefaultMiddleware(config *configuration.Configuration) (changed bool, err error) {
+	// Default to openshift middleware for relevant types
+	// This allows custom configs based on old default configs to continue to work
+	if config.Middleware == nil {
+		config.Middleware = map[string][]configuration.Middleware{}
+	}
+	for _, middlewareType := range []string{"registry", "repository", "storage"} {
+		if _, exists := config.Middleware[middlewareType]; !exists {
+			config.Middleware[middlewareType] = []configuration.Middleware{}
+		}
+		found := false
+		for _, middleware := range config.Middleware[middlewareType] {
+			if middleware.Name == "openshift" {
+				found = true
+				break
+			}
+		}
+		if found {
+			continue
+		}
+		config.Middleware[middlewareType] = append(config.Middleware[middlewareType], configuration.Middleware{
```

Comment:
`err` is not returned in any case here.  How about we record any time we default something on this line and return that so that if we've changed something we can give them a message that says their config is obsolete and here is what we added to get it up to date?

LGTM otherwise

and tests :smile: